### ISSUE TITLE
photospline - failing tests causing a cascade of other test failures (Trac #1813)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1805]
[source:IceCube/projects/photospline photopslne] is failing specifically in

  [source:IceCube/projects/photospline/trunk/private/lib/bspline.c#L195] and
  [source:IceCube/projects/photospline/trunk/private/lib/bspline_multi.c#L180].

This is causing test failures in at least photonics-service, spline-reco, and lilliput. This bug was possibly introduced in r148888/IceCube, and has been hidden by some bad shell logic in that particular buildstep causing a green false positive. (That shell logic has been fixed)

Migrated from https://code.icecube.wisc.edu/ticket/1813
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:12:38", 
    "description": "[source:IceCube/projects/photospline photopslne] is failing specifically in\n\n  [source:IceCube/projects/photospline/trunk/private/lib/bspline.c#L195] and\n  [source:IceCube/projects/photospline/trunk/private/lib/bspline_multi.c#L180].\n\nThis is causing test failures in at least photonics-service, spline-reco, and lilliput. This bug was possibly introduced in r148888/IceCube, and has been hidden by some bad shell logic in that particular buildstep causing a green false positive. (That shell logic has been fixed)", 
    "reporter": "nega", 
    "cc": "mwallraff, dunkman, jacob.feintzeig, sybenzvi, olivas", 
    "resolution": "fixed", 
    "_ts": "1550067158057333", 
    "component": "combo reconstruction", 
    "summary": "photospline - failing tests causing a cascade of other test failures", 
    "priority": "critical", 
    "keywords": "", 
    "time": "2016-08-04T15:25:30", 
    "milestone": "", 
    "owner": "jvansanten", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160805T16:25:22: 
* olivas changed priority from "normal" to "critical"
* olivas commented:

> The commit r148888 was backed out in r149075.  Reversing it did, in fact, allow the photospline test to pass locally for me.  Let's see what the bots say...

**Comment by icecube-trac** [comment]
Trac update at 20160815T08:44:14: 
* jvansanten changed status from "new" to "closed"
* jvansanten changed resolution from "" to "fixed"
* jvansanten commented:

> In changeset 149245:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="149245"
> Add annotations to keep static analysis happy. Fixes #1805 and #1776.
> 
> Also, add a note to the next person who forgets about the tricks
> employed to avoid special-casing evaluations outside the full support
> of the spline.
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:12:38: 
* olivas changed milestone from "August 2016" to ""
* olivas commented:

> Milestone August 2016 deleted
